### PR TITLE
inclusive_start/inclusive_end options for Timex.between?

### PR DESCRIPTION
**Pull request by glassbead0**
* Timex.between? the :inclusive option can now take `:start`, `:end` in addition to `true`
### Summary of changes

I have a use case to ask if a given time is between two other times, with one bound being inclusive, and the other being exclusive. I imagine this is useful for anyone iterating over adjacent intervals of time.

### Checklist

- [x] New functions have typespecs, changed functions were updated
- [x] Same for documentation, including moduledocs
- [x] Tests were added or updated to cover changes
- [x] Commits were squashed into a single coherent commit

### CODE REVIEW

File: lib/timex.ex [hunk 1]
```diff
@@ -771,26 +771,33 @@ defmodule Timex do
...
-  by setting the `:inclusive` option to `true`.
+  by setting the `:inclusive` option to `true`. You can also specify `:inclusive_start` and/or
+  `:inclusive_end` options, which will override the `:inclusive` option for the given bound.
 
   """
   @type between_options :: [
-    inclusive: boolean
+    inclusive: boolean,
+    inclusive_start: boolean,
+    inclusive_end: boolean
```
Comment (pezra) [inline lib/timex.ex]: it might be nicer to expand the domain of `inclusive` rather than introducing new options. for example:

```
iex> Timex.between?(t1, t2, t3, inclusive: true)
iex> Timex.between?(t1, t2, t3, inclusive: :start)
iex> Timex.between?(t1, t2, t3, inclusive: :end)
```
that would avoid some awkward potential situations where `inclusive` and `inclusive_start` and `inclusive_end` disagree with one another.

```
iex> Timex.between?(t1, t2, t3, inclusive: false, inclusive_start: true)
iex> Timex.between?(t1, t2, t3, inclusive: true, inclusive_start: false)
```

File: lib/timex.ex [hunk 2]
```diff
@@ -770,27 +770,39 @@ defmodule Timex do
   Returns a boolean indicating whether the first `Timex.Comparable` occurs between the second
   and third.
 
-  By default, the `start`and `ending` bounds are *exclusive*. You can opt for inclusive bounds
-  by setting the `:inclusive` option to `true`.
+  By default, the `start`and `ending` bounds are *exclusive*. You can opt for inclusive `start`
+  and `end` bounds by setting the `:inclusive` option to `true`. You can also specify
+  `:inclusive_start` and/or `:inclusive_end` options, which will override the `:inclusive`
+  option for the given bound.
```
Comment (ckhrysze) [inline lib/timex.ex]: I like the updated option! My only ask here is that the doc reflect the latest code update and consistent use of either 'end' or 'ending'.

Comment (glassbead0) [reply lib/timex.ex]: 👍 Thanks for the reminder. I've updated the doc.

File: lib/timex.ex [hunk 3]
```diff
@@ -770,27 +770,38 @@ defmodule Timex do
...
     inclusive: boolean
+    | :start
+    | :end
   ]
   @spec between?(Time, Time, Time, between_options) :: boolean | {:error, term}
   @spec between?(Comparable.comparable, Comparable.comparable, Comparable.comparable, between_options) ::
     boolean | {:error, term}
   def between?(a, start, ending, options \\ []) do
-    inclusive = Keyword.get(options, :inclusive, false)
+    [inclusive_start, inclusive_end] =
```
Comment (bitwalker) [inline lib/timex.ex]: @ckhrysze Can we make this a tuple rather than a list? I missed the review on this, sorry. As a rule I prefer tuples for a known-size set of things, with small N. It better reflects the purpose, access is constant, and is a smaller in-memory structure (though obviously for cases like this, that isn't really important). You can commit the change directly to master, no need to go through another PR.
